sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/TaskKiller.java

```diff
@@ -74,25 +81,44 @@ public static void killTask(TaskID taskId) {
             return;
         }
 
```

Comment:
Having a log statement before lock acquisition would be good IMO. It isn't _actually_ possible to achieve a deadlock here, but I'm always leery of having locks that are "silently" acquired.